mqtt_recorder/recorder.py

```diff
@@ -86,11 +101,8 @@ def stop_recording(self):
         self.__client.loop_stop()
         logger.info('Recording stopped')
         self.__recording = False
-        logger.info('Saving messages to output file')
-        with open(self.__file_name, 'w', newline='') as csvfile:
-            writer = csv.writer(csvfile)
-            for message in self.__messages:
-                writer.writerow(message)
+        self.__messages.put([])
```

Comment:
Is this really necessary here?

Reply:
Yes, you need to signal the thread to stop writing so it can be properly joined in `stop_recording()`.

I opted for an empty list, but really, it can be anything considered False when checking at `__csv_writer`.

Reply:
Oh I see, completely ingnored the break in the writing thread. Thanks!